### ISSUE TITLE
iOS cookies not persistent after app exit and restart unlike Android

### ISSUE THREAD

**Issue by narayanncube** [apache/cordova-plugin-inappbrowser#509]
# Bug Report

## Problem
In iOS cookies are not persistent once app is exited and restarted. This creates problem with hosted web app saving cookies or local storage.

On Android it works

### What is expected to happen?
It should save cookies and local storage on IOS

### What does actually happen?
It doesn't save cookies and local storage.

### Command or Code

var ref = cordova.InAppBrowser.open('http://www.syncaridebeta.com', '_blank', 'location=no,toolbar=no,usewkwebview=yes,clearcache=no,clearsessioncache=no,cleardata=no,disallowoverscroll=yes');

## Checklist

- [ ] I searched for existing GitHub issues
- [ ] I updated all Cordova tooling to most recent version
- [ ] I included all the necessary information above

**Comment by janpio** [comment]
Could you please confirm if this is also the case when using `usewkwebview=no`?

**Comment by narayanncube** [comment]
> Could you please confirm if this is also the case when using `usewkwebview=no`?

Yes. Its the case. Just tried with both options. Tried following https://github.com/ionic-team/cordova-plugin-ionic-webview/issues/22 for any solution and specifically tried this https://github.com/CWBudde/cordova-plugin-wkwebview-inject-cookie but without any success. May be i was placing in code in wrong place or namespace is wrong or may be ionic thats why

**Comment by janpio** [comment]
Is this reproducible in a new, plain Cordova app? A minimal reproduction repository would really help to debug and later fix this issue. More information on how to create one: https://github.com/apache/cordova-contribute/blob/master/create-reproduction.md

**Comment by alebupal** [comment]
Any solution? I have the same problem.

**Comment by BradCB** [comment]
Any solution? I have the same problem.

**Comment by michaellavender** [comment]
Per my comment above, this actually was a problem with how we were creating the cookies server-side. They were being created as session cookes (without an expiration date), so would expire as soon as the app was closed because each instance of the app creates a separate inappbrowser instance. By setting an explicit expiration date on the cookie returned by the server this causes the browser to treat it as persistent, rather than for just a single session, and it's there for subsequent instances of the app. In our case we have no need for the cookie to ever expire so we're setting it to have a 12/31/9999 expiration when returned in the response from the server.

**Comment by jorgedelcampo** [comment]
Hi, how do you resolved this? I'm trying to start session or set cookies but i can't with Phonegap. If i try from server with any browser session set ok.

**Comment by Jack-Rogers** [comment]
@michaellavender 
While I'm not using Cordova, I had the exact same issue using WKWebView in swift. Honestly, I tried everything and then saw your comment, I can't thank you enough! It's such a stupid issue, I know of no other implementations that treat a cookie without expiration like this! 
Of course, the only one that would, would be Apple 🙄

**Comment by batmanbury** [comment]
> Yes. Its the case. Just tried with both options. Tried following [ionic-team/cordova-plugin-ionic-webview#22](https://github.com/ionic-team/cordova-plugin-ionic-webview/issues/22) for any solution and specifically tried this https://github.com/CWBudde/cordova-plugin-wkwebview-inject-cookie but without any success. May be i was placing in code in wrong place or namespace is wrong or may be ionic thats why

@narayanncube Did you find a way to explicitly set the cookie expiration? I'm trying to find how I might do the same. I'm currently using the same plugin you mentioned to inject a cookie, so I'm trying to find how to either update this inject-cookie plugin to set the expiration there, or to just get rid of the plugin, and inject the cookie manually and set the expiration myself...
